File: scriptedmain.py
#!/usr/bin/env python

import os, sys

def main():
	print "Directory: " + os.getcwd()

	print "Program: " + sys.argv[0]

	print "Number of Args: %d" % len(sys.argv)

	for arg in sys.argv:
		print "Arg: " + arg

if __name__=="__main__": main()

File: scriptedmain.pl
#!/usr/bin/env perl

use strict;
use Cwd qw(getcwd);

sub main {
	print "Dirctory: " . getcwd . "\n";

	print "Program: $0\n";

	print "Number of Args: " . ($#ARGV + 1) . "\n";

	foreach my $i (0 .. $#ARGV) {
		print "Arg: $ARGV[$i]\n";
	}
}

unless(caller) { main; }